server/src-lib/Hasura/GraphQL/Execute.hs

```diff
@@ -357,46 +357,41 @@ execRemoteGQ
   -> m (HttpResponse EncJSON)
 execRemoteGQ reqId userInfo reqHdrs q rsi opDef = do
   execCtx <- ask
-  let logger  = _ecxLogger execCtx
+  let logger = _ecxLogger execCtx
       manager = _ecxHttpManager execCtx
-      opTy    = G._todType opDef
+      opTy = G._todType opDef
   when (opTy == G.OperationTypeSubscription) $
     throw400 NotSupported "subscription to remote server is not supported"
   hdrs <- getHeadersFromConf hdrConf
-  let confHdrs   = map (\(k, v) -> (CI.mk $ CS.cs k, CS.cs v)) hdrs
+  let confHdrs = map (\(k, v) -> (CI.mk $ CS.cs k, CS.cs v)) hdrs
       clientHdrs = bool [] filteredHeaders fwdClientHdrs
       -- filter out duplicate headers
       -- priority: conf headers > resolved userinfo vars > client headers
-      hdrMaps    = [ Map.fromList confHdrs
-                   , Map.fromList userInfoToHdrs
-                   , Map.fromList clientHdrs
-                   ]
-      finalHdrs  = foldr Map.union Map.empty hdrMaps
-      options    = wreqOptions manager (Map.toList finalHdrs)
-
+      hdrMaps =
+        [ Map.fromList confHdrs
+        , Map.fromList userInfoToHdrs
+        , Map.fromList clientHdrs
+        ]
+      finalHdrs = foldr Map.union Map.empty hdrMaps
+      options = wreqOptions manager (Map.toList finalHdrs) (Just $ timeout * 1000000)
```

Comment:
Why not pass seconds to `wreqOptions` and do the conversion there?

Reply:
because `wreqOptions` is a library function and you may need more granularity than seconds for that.